opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -288,8 +288,10 @@ namespace Opm
             from = WellProducerCMode2String(ws.production_cmode);
         }
         const bool oscillating = std::count(this->well_control_log_.begin(), this->well_control_log_.end(), from) >= this->param_.max_number_of_well_switches_;
-
-        if (oscillating || this->wellUnderZeroRateTarget(simulator, well_state, deferred_logger) || !(this->well_ecl_.getStatus() == WellStatus::OPEN)) {
+        const int iterationIdx = simulator.model().newtonMethod().numIterations();
+        const int episodeIdx = simulator.episodeIndex();
+        const int nupcol = schedule[episodeIdx].nupcol();
+        if ( (oscillating && iterationIdx > nupcol)|| this->wellUnderZeroRateTarget(simulator, well_state, deferred_logger) || !(this->well_ecl_.getStatus() == WellStatus::OPEN)) {
```

Comment:
>= nupcol?

Reply:
In updateWellControl(..) it is iterationIdx >nupcol. This makes it consistent. But I don't know if waiting one iteration is on purpose or not.

Reply:
You mean what was introduced in https://github.com/OPM/opm-simulators/pull/5753? Since we start counting iterations from 0 I believe we should use \>= unless there is a clear (and documented) reason to do so.

Reply:
Since I made #5773, I assume > is a bug, I will update both instances to >= and retest.

Reply:
Thanks. Can be discussed if it is a bug, but prior to https://github.com/OPM/opm-simulators/pull/6100 it was a mix between \>= and \>, and better to be consistent. Apparently I missed the one in updateWellControls in that PR..